src/main/webapp/3Dscript.js

```diff
@@ -45,144 +49,285 @@ function initThreeJs() {
-  material = new THREE.MeshBasicMaterial({color: 'red'});
-  geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
-  orientation = new THREE.InstancedMesh(geometry, material, pose.length);
-  orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
-  scene.add(orientation);
   plotOrientation();
 }
 
+
 /**
- * This uses the pose data to create a blue line representing the pose
- * trajectory. 
+ * Creates all the children for each axis of orientation. Each object
+ * is a geometry instance, initializing the object with one geometry
+ * call rather than the number of data points. (O(1) vs. O(N))
+ */
+function createInstances() {
+  // Blue corresponds to the Z axis.
+  let material = new THREE.MeshBasicMaterial({color: 'blue'});
+  const geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
+  orientation.Z = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.Z.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.Z);
+
+  // Green corresponds to the Z axis.
+  material = new THREE.MeshBasicMaterial({color: 'green'});
+  orientation.Y = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.Y.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.Y);
+
+  // Red corresponds to the Z axis.
+  material = new THREE.MeshBasicMaterial({color: 'red'});
+  orientation.X = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.X.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.X);
+}
+/**
+  * Converts latitude, longitude, and altitude coordinates (LLA) to
+  * Earth-Centered, Earth-Fixed coordinates (ECEF).
+  * @param {double} lat Latitude represented in Decimal Degrees
+  * @param {double} lng Longitude represented in Decimal Degrees
+  * @param {double} alt Altitude represented in Decimal Degrees
+  * @return {array} An array containing x, y, z ECEF coordinates.
+  */
+function llaToEcef(lat, lng, alt) {
+  // Constants.
+  const majorAxis = 6378137;
+  const flatConstant = 1 / 298.25722356;
+  const minorAxis = majorAxis*(1-flatConstant);
+  // Eccentricity.
+  const eSq = (Math.pow(majorAxis, 2) -
+    Math.pow(minorAxis, 2)) / Math.pow(majorAxis, 2);
+  const lambda = THREE.Math.degToRad(lat);
+  const phi = THREE.Math.degToRad(lng);
+  const radius = majorAxis /
+    Math.sqrt(1 - eSq * Math.pow(Math.sin(lambda), 2));
+  // Conversion.
+  const x = (radius + alt) * Math.cos(lambda) * Math.cos(phi);
+  const y = (radius + alt) * Math.cos(lambda) * Math.sin(phi);
+  const z = ((Math.pow(minorAxis, 2) * radius) /
+    Math.pow(majorAxis, 2) + alt) * Math.sin(lambda);
+
+  return [x, y, z];
+}
+
+/**
+ * Converts Earth-Centered, Earth-Fixed coordinates (ECEF) to
+ * East-North-Up coordinates (ENU).
+ * @param {double} x Latitude represented in ECEF coordinates.
+ * @param {double} y Longitude represented in ECEF coordinate.
+ * @param {double} z Altitude represented in ECEF coordinates.
```

Comment:
ECEF does not have lat/lng/alt. It's just x/y/z (probably in meters). An increase in x, for example, can correspond to an increase in altitude in one place on the globe, a decrease in altitude on the opposite side of the globe, and neither elsewhere.

Reply:
Can you use the proper types for JSDoc comments, while double is a more exact description of the type, it is not a valid JSDoc type which means we can't run static type checking using these comments? Instead you can put @param {number} and specify that it is type double in the comment. That way we can have static type linting to prevent as many runtime bugs. Not a huge deal either way, just thought I'd mention it.
https://jsdoc.app/tags-type.html